staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -147,6 +147,11 @@ func (s *store) Create(ctx context.Context, key string, obj, out runtime.Object,
 	if version, err := s.versioner.ObjectResourceVersion(obj); err == nil && version != 0 {
 		return errors.New("resourceVersion should not be set on objects to be created")
 	}
+	// ensure that the serialized version does not contain any other fields that are not allowed
+	// to be stored at rest.
```

Comment:
the fact that resourceVersion is stored out of band is an implementation detail specific to etcd storage